src/cpp/dds/publisher/PublisherImpl.cpp

```diff
@@ -0,0 +1,492 @@
+    w_att.endpoint.remoteLocatorList = att_.remoteLocatorList;
+    w_att.mode = writer_qos.m_publishMode.kind == SYNCHRONOUS_PUBLISH_MODE ? SYNCHRONOUS_WRITER : ASYNCHRONOUS_WRITER;
+    w_att.endpoint.properties = att_.properties;
+
+    if (att_.getEntityID() > 0)
+    {
+        w_att.endpoint.setEntityID(static_cast<uint8_t>(att_.getEntityID()));
+    }
+
+    if (att_.getUserDefinedID() > 0)
+    {
+        w_att.endpoint.setUserDefinedID(static_cast<uint8_t>(att_.getUserDefinedID()));
+    }
+
+    w_att.times = att_.times;
+    w_att.liveliness_kind = writer_qos.m_liveliness.kind;
+    w_att.liveliness_lease_duration = writer_qos.m_liveliness.lease_duration;
+    w_att.matched_readers_allocation = att_.matched_subscriber_allocation;
+
+    // TODO(Ricardo) Remove in future
+    // Insert topic_name and partitions
+    Property property;
+    property.name("topic_name");
+    property.value(topic_att.getTopicName().c_str());
+    w_att.endpoint.properties.properties().push_back(std::move(property));
+
+    if(writer_qos.m_partition.getNames().size() > 0)
+    {
+        property.name("partitions");
+        std::string partitions;
+        for(auto partition : writer_qos.m_partition.getNames())
+        {
+            partitions += partition + ";";
+        }
+        property.value(std::move(partitions));
+        w_att.endpoint.properties.properties().push_back(std::move(property));
+    }
+
+    if (writer_qos.m_disablePositiveACKs.enabled &&
+        writer_qos.m_disablePositiveACKs.duration != c_TimeInfinite)
+    {
+        w_att.disable_positive_acks = true;
+        w_att.keep_duration = writer_qos.m_disablePositiveACKs.duration;
+    }
+
+    DataWriterImpl* impl = new DataWriterImpl(
+        this,
+        type_support,
+        topic_att,
+        w_att,
+        writer_qos,
+        att_.historyMemoryPolicy,
+        listener);
+
+    if(impl->writer_ == nullptr)
+    {
+        logError(PUBLISHER, "Problem creating associated Writer");
+        delete impl;
+        return nullptr;
+    }
+
+    DataWriter* writer = new DataWriter(impl);
+    impl->user_datawriter_ = writer;
+
+    //REGISTER THE WRITER
+    rtps_participant_->registerWriter(impl->writer_, topic_att, writer_qos);
+
+    {
+        std::lock_guard<std::mutex> lock(mtx_writers_);
+        writers_[topic_att.getTopicDataType().to_string()].push_back(impl);
+    }
+
+    return writer;
+}
+
+bool PublisherImpl::delete_datawriter(
+        DataWriter* writer)
+{
+    std::lock_guard<std::mutex> lock(mtx_writers_);
+    auto vit = writers_.find(writer->get_topic().getTopicName().to_string());
+    if (vit != writers_.end())
+    {
+        auto dw_it = std::find(vit->second.begin(), vit->second.end(), writer->impl_);
+        if (dw_it != vit->second.end())
+        {
+            vit->second.erase(dw_it);
+            return true;
+        }
+    }
+    return false;
+}
+
+DataWriter* PublisherImpl::lookup_datawriter(
+    const std::string& topic_name) const
```

Comment:
Coding style. Indent with 8 spaces
```suggestion
        const std::string& topic_name) const
```